kore/test/Test/Kore/Reachability/Prove.hs

```diff
@@ -69,308 +61,127 @@ import Test.Tasty.HUnit.Ext
+                    -- second step, which does not run here.
+                    let stuck = mkSimpleClaim Mock.b Mock.b
+                        expect = MultiAnd.singleton (StuckClaim stuck)
+                    assertEqual "" expect actual
+        in
+        [ mkTest "OnePath" simpleOnePathClaim
+        , mkTest "AllPath" simpleAllPathClaim
+        ]
+    , testGroup "proves direct implication" $
+        let mkTest name mkSimpleClaim =
+                testCase name $ do
+                    actual <- proveClaims_
+                        Unlimited
+                        (Limit 1)
+                        []
+                        [mkSimpleClaim Mock.a Mock.a]
+                        []
+                    assertEqual "" MultiAnd.top actual
+        in
+        [ mkTest "OnePath" simpleOnePathClaim
+        , mkTest "AllPath" simpleAllPathClaim
+        ]
+    , testGroup "does not prove implication without rewriting" $
+        let mkTest name mkSimpleClaim =
+                testCase name $ do
+                    let claim = mkSimpleClaim Mock.a Mock.b
+                    actual <- proveClaims_
+                        Unlimited
+                        Unlimited
+                        []
+                        [claim]
+                        []
+                    let expect = MultiAnd.singleton (StuckClaim claim)
+                    assertEqual "" expect actual
+        in
+        [ mkTest "OnePath" simpleOnePathClaim
+        , mkTest "AllPath" simpleAllPathClaim
+        ]
+    , testGroup "proves by rewriting into disjunction" $
+        let mkTest name mkSomeClaim =
+                testCase name $ do
+                    actual <- proveClaims_
+                        (Limit 2)
+                        Unlimited
+                        [ simpleAxiom Mock.a Mock.b ]
+                        [ mkSomeClaim
+                            (Pattern.fromTermLike Mock.a)
+                            (OrPattern.fromPatterns $ map Pattern.fromTermLike
+                                [ Mock.b
+                                , Mock.c
+                                ]
+                            )
+                            []
+                        ]
+                        []
+                    assertEqual "" MultiAnd.top actual
+        in
+        [ mkTest "OnePath" mkSomeClaimOnePath
+        , mkTest "AllPath" mkSomeClaimAllPath
+        ]
+    , testGroup "proves anything with cyclic rules" $
+        let mkTest name mkSimpleClaim =
+                testCase name $ do
+                    actual <- proveClaims_
+                        Unlimited
+                        (Limit 3)
+                        [ simpleAxiom Mock.a Mock.a ]
+                        [ mkSimpleClaim Mock.a Mock.b ]
+                        []
+                    assertEqual "" MultiAnd.top actual
+        in
+        [ mkTest "OnePath" simpleOnePathClaim
+        , mkTest "AllPath" simpleAllPathClaim
+        ]
```

Comment:
Any reason why we don't want the mixed cases anymore?

Reply:
The only thing that these mixed cases actually tested is that the backend doesn't try to use one-path claims to prove all-path claims, or vice versa. There are already tests specifically for that, and I didn't think it made sense to duplicate that check too many times.